### ISSUE TITLE
[.NET 5 RC1] "HttpRequestException: Address family not supported by protocol (127.0.0.1:9001)" thrown when Lambda invoked

### ISSUE THREAD

**Issue by martincostello** [aws/aws-lambda-dotnet#730]
## Description

I have a [Lambda function](https://github.com/martincostello/alexa-london-travel) that uses the .NET Core custom runtime via the `Amazon.Lambda.RuntimeSupport` NuGet package, so with [the release of .NET 5.0 RC1 as a "go live" release](https://devblogs.microsoft.com/dotnet/announcing-net-5-0-rc-1/) I updated it to target `net5.0` using RC1 to try it out.

The tests [passed as expected in CI](https://travis-ci.org/github/martincostello/alexa-london-travel/builds/727188410), however when the Lambda was deployed to the Lambda environment it failed with the exception:

```
System.Net.Http.HttpRequestException: Address family not supported by protocol (127.0.0.1:9001)
```

The error comes from the HTTP call made here:

https://github.com/aws/aws-lambda-dotnet/blob/c306036773617d3e2b81dc9d35c9a687b9db966f/Libraries/src/Amazon.Lambda.RuntimeSupport/Client/InternalClientAdapted.cs#L233

## Reproduction Steps

Clone the Lambda function from https://github.com/martincostello/alexa-london-travel/commit/d5d761816805fd0c6f2f8ad7b40c76fd036706db and deploy to AWS Lambda and invoke the function.

### Logs

```
  | 2020-09-14T23:10:37.855+01:00 | Unhandled exception. System.Net.Http.HttpRequestException: Address family not supported by protocol (127.0.0.1:9001)
  | 2020-09-14T23:10:37.855+01:00 | ---> System.Net.Sockets.SocketException (97): Address family not supported by protocol
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Sockets.Socket..ctor(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Connections.SocketsConnectionFactory.CreateSocket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType, EndPoint endPoint, IConnectionProperties options)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Connections.SocketsConnectionFactory.ConnectAsync(EndPoint endPoint, IConnectionProperties options, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.ConnectHelper.ConnectAsync(ConnectionFactory factory, DnsEndPoint endPoint, IConnectionProperties options, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | --- End of inner exception stack trace ---
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.ConnectHelper.ConnectAsync(ConnectionFactory factory, DnsEndPoint endPoint, IConnectionProperties options, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at Amazon.Lambda.RuntimeSupport.InternalRuntimeApiClient.NextAsync(CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at Amazon.Lambda.RuntimeSupport.RuntimeApiClient.GetNextInvocationAsync(CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at Amazon.Lambda.RuntimeSupport.LambdaBootstrap.InvokeOnceAsync(CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at Amazon.Lambda.RuntimeSupport.LambdaBootstrap.RunAsync(CancellationToken cancellationToken)
  | 2020-09-14T23:10:37.855+01:00 | at MartinCostello.LondonTravel.Skill.FunctionEntrypoint.RunAsync(HttpClient httpClient, CancellationToken cancellationToken) in /_/src/LondonTravel.Skill/FunctionEntrypoint.cs:line 42
  | 2020-09-14T23:10:37.855+01:00 | at MartinCostello.LondonTravel.Skill.FunctionEntrypoint.Main() in /_/src/LondonTravel.Skill/FunctionEntrypoint.cs:line 52
  | 2020-09-14T23:10:37.855+01:00 | at MartinCostello.LondonTravel.Skill.FunctionEntrypoint.<Main>()
  | 2020-09-14T23:10:37.939+01:00 | START RequestId: e9f44aa6-1849-46e8-9fd7-60b2cb95620b Version: $LATEST
```

While not yet supported by AWS Lambda itself, I thought I'd raise this here in the first instance in case this is an issue that stems from the Lambda runtime. If it is not, then I can open the issue against the [dotnet/runtime repo](https://github.com/dotnet/runtime) for investigation.

### Environment

* Build Version: 1.1.1
* OS Info: AmazonLinux
* Build Environment: `dotnet` SDK
* Targeted .NET Platform:  

## Resolution

The Lambda successfully receives requests to process from the Lambda runtime.

---

This is a :bug: bug-report

**Comment by normj** [comment]
Interesting that it does work correctly with .NET 5 preview 8 which was what I lasted tested and just confirmed it does still work with preview 8. I get the same error for RC1.

**Comment by normj** [comment]
I do think this is regression with .NET 5 RC. I have opened an issue on their repository to work with them to get the issue fixed or come up with a work around. https://github.com/dotnet/runtime/issues/42233

**Comment by normj** [comment]
I confirmed using the nightly builds of .NET 5 RC2 the issue is fixed. So you will either have to use the nightly builds or wait till RC2 is released. Thanks so much for alerting us of the issue.

**Comment by martincostello** [comment]
Awesome - thanks for the detective work Norm 🕵🏻‍♂️

I think I'll hold out for RC2 for now.

**Comment by martincostello** [comment]
FYI the fix came about here: https://github.com/dotnet/runtime/pull/41955#discussion_r485276498

**Comment by martincostello** [comment]
FYI I've confirmed the fix myself via updating my Lambda to use `5.0.100-rc.2.20464.14` (https://github.com/martincostello/alexa-london-travel/pull/275).

**Comment by ashishdhingra** [comment]
Closing based on last comment.
